internal/daemon/controller/handler.go

```diff
@@ -737,3 +738,17 @@ func wrapHandlerWithCallbackInterceptor(h http.Handler, c *Controller) http.Hand
 		h.ServeHTTP(w, req)
 	})
 }
+
+// getActions takes in a URL and returns the actions from the URL
+func getActions(url string) []string {
+	parts := strings.Split(url, "/")
+	lastPart := parts[len(parts)-1]
+
+	items := strings.Split(lastPart, ":")
+
+	if len(items) > 1 {
+		return items[1:]
+	}
+
+	return []string{}
+}
```

Comment:
Arguably for a URL like `/v1/auth-methods/amoidc_1234567890:authenticate:callback` this should return both `authenticate` and `callback`, no? Could you add some tests for this function?